### PR TITLE
Add python support to ITT API

### PR DESCRIPTION
**Pull request by eparshut**
Add Python binding to ITT API.
Many kudos to @esuldin for the work done and his contribution.

### CODE REVIEW

**Review by abrown — COMMENTED** [review]
@eparshut asked me to take a look at this so here are some thoughts:
- First, I think this is great! Having control over collection in more languages is extremely helpful; I wasn't even aware this existed until I saw it upstreamed here.
- A _naming_ nit: I would probably go with `python` as the top-level directory for this code to match the `rust` directory. And I would have expected the PyPI package to be named `ittapi`, but it's probably too late to change that.
- I noticed different coding styles throughout; maybe we should add a formatter pass in CI
- Which brings me to CI: this addition would really benefit from building itself in a GitHub workflow, running the tests, running lints, etc.

File: pyitt/LICENSE
Comment (esuldin) [inline pyitt/LICENSE]: The licenses that are already used for ittapi do not cover the pyitt code. Therefore, I'm not fully sure that the license for pyitt can be simply removed.

File: pyitt/pyproject.toml
Comment (esuldin) [inline pyitt/pyproject.toml]: The usage of Intel email address is not align with Own-Time Open Source Project Approval that pyitt has. Therefore, please return the original email address back.

**Review by esuldin — CHANGES_REQUESTED** [review]

**Comment by lubomirov** [comment]
1. I would recommend to use `python` for the subfolder and not `pyitt`, but this decision is up to @eparshut , as component owner.
2. @esuldin, is there any chance to create and deploy package automatically with python support?
3. This code review is Egor's contribution into Intel belongs component `ittapi`. Consequently, Egor is co-author of the `ittapi`. There shouldn't be any references to `pyitt` project, including the different licenses in subfolders, original author emails, etc. This is a new code with new contributor

File: pyitt/pyproject.toml
Comment (lubomirov) [reply pyitt/pyproject.toml]: If this is a script of "deployment" than it should be `ittapi` at the line number 2, the name of component owner and co-authors with their emails.

File: pyitt/LICENSE
Comment (lubomirov) [reply pyitt/LICENSE]: I doubt that we should keep one more license in `ittapi` component. Moreover, I think, one component = one license in the root. But Egor is right and we must include his name/copyright about him into generalized code ownership. I don't know what the practice of doing this?

Comment (oburova-intel) [reply pyitt/LICENSE]: Do we have any legal guidance/confirmation on this? 
_But Egor is right and we must include his name/copyright about him into generalized code ownership. I don't know what the practice of doing this?_

I believe due to sanctions we can't use yandex.ru email addresses in our source code

File: pyitt/pyproject.toml
Comment (oburova-intel) [reply pyitt/pyproject.toml]: I believe Egor is an Intel employee and copywright should be owned by Intel anyway

File: python/pyproject.toml
Comment (lubomirov) [inline python/pyproject.toml:2]: should be `ittapi` by analogue with RUST naming schema for publishing

**Review by lubomirov — CHANGES_REQUESTED** [review]

File: python/pyproject.toml
Comment (eparshut) [inline python/pyproject.toml:1]: It is need to add to CI workflow build and test steps for python bindings before merge changes to master to prevent potential future problem

Comment (Andy51) [reply python/pyproject.toml:1]: I think this could be done as a separate step. At this moment the build/test can be verified manually

File: pyitt/pyproject.toml
Comment (esuldin) [reply pyitt/pyproject.toml]: For projects with Own-Time Open Source Project Approval, the copyright is held by their authors.

**Comment by esuldin** [comment]
>     4. Let's point Egor's contribution here through his Intel's email - `egor.suldin@intel.com`

The usage of Intel email address is not align with Own-Time Open Source Project Approval that pyitt has. Therefore, please use original email address or just remove it at all from pyproject.toml.

File: python/setup.py
Comment (lubomirov) [inline python/setup.py]: Unchanged `intel-ittapi` project name still appears in Readme.md in several places

**Review by lubomirov — CHANGES_REQUESTED** [review]

File: python/ittapi.native/collection_control.cpp
Comment (eparshut) [inline python/ittapi.native/collection_control.cpp:6]: Need to rename `pyitt` namespace to `ittapi` in all files

File: python/utest/pyitt_native_mock/__init__.py
Comment (eparshut) [inline python/utest/pyitt_native_mock/__init__.py:4]: It would also be nice to rename all `pyitt` namings in utest layer to `ittapi`

File: python/pyproject.toml
Comment (lalithsharan-intel) [reply python/pyproject.toml]: changed to ittapi

File: python/README.md
Comment (lalithsharan-intel) [reply python/README.md]: changed to ittapi

File: python/setup.py
Comment (lalithsharan-intel) [reply python/setup.py]: changed

File: python/ittapi.native/collection_control.cpp
Comment (lalithsharan-intel) [reply python/ittapi.native/collection_control.cpp]: changed

File: python/utest/pyitt_native_mock/__init__.py
Comment (lalithsharan-intel) [reply python/utest/pyitt_native_mock/__init__.py]: changed

File: python/README.md
Comment (lalithsharan-intel) [reply python/README.md]: changed

**Review by lubomirov — APPROVED** [review]

**Review by lalithsharan-intel — APPROVED** [review]

**Review by Andy51 — APPROVED** [review]

**Pull request merged**
